platform/kubernetes/kubernetes.go

```diff
@@ -79,43 +76,35 @@ func isAddon(obj namespacedLabeled) bool {
 
 // --- /add ons
 
+type Applier interface {
+	Delete(logger log.Logger, def []byte) error
+	Create(logger log.Logger, def []byte) error
+	Apply(logger log.Logger, def []byte) error
```

Comment:
-counter is that you can absorb that logical distinction by defining sync.Apply an "upsert" for everything, always, even if that means extra code (although we cannot really do it properly for daemonsets at the minute).

In other words, although `kubectl apply` does not do the same thing if the resource exists or doesn't exist, `SyncAction.Apply` can. Try to. Where possible.